ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -823,8 +824,9 @@ IndexValue getIndexValueOfLastPut() throws StoreException {
    * Maps the segment of index either as a memory map or a in memory buffer depending on config.
    * @throws StoreException if there are problems with the index
    */
-  private void map() throws StoreException {
+  private void map() throws StoreException, FileNotFoundException {
     rwLock.writeLock().lock();
+    checkDataIntegrity(indexFile);
```

Comment:
I would recommend to directly use `serEntries` initialized below to create a CrcInputStream for data integrity validation. This is good for two reasons: 1. We don't have to read the file twice(although the second read would be just a memory copy) 2. We can deal with FileNotFoundException in the same manner.

Reply:
Sure,  I use `serEntries` to check data integrity in latest commit. For non-heap buffer, I don't find a good way to avoid bytes transfer (as you can see I still transfer bytes to a heap bytes array for crc calculation)